eZ/Publish/Core/Search/Common/FieldValueMapper/FloatMapper.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version //autogentag//
+ */
+namespace eZ\Publish\Core\Search\Common\FieldValueMapper;
+
+use eZ\Publish\Core\Search\Common\FieldValueMapper;
+use eZ\Publish\SPI\Search\FieldType\FloatField;
+use eZ\Publish\SPI\Search\Field;
+
+/**
+ * Common float field value mapper implementation.
+ */
+class FloatMapper extends FieldValueMapper
+{
+    /**
+     * Check if field can be mapped.
+     *
+     * @param \eZ\Publish\SPI\Search\Field $field
+     *
+     * @return bool
+     */
+    public function canMap(Field $field)
+    {
+        return $field->type instanceof FloatField;
+    }
+
+    /**
+     * Map field value to a proper search engine representation.
+     *
+     * @param \eZ\Publish\SPI\Search\Field $field
+     *
+     * @return mixed
+     */
+    public function map(Field $field)
+    {
+        return $this->fixupFloat($field->value);
+    }
+
+    /**
+     * Convert to a proper search engine representation.
+     *
+     * @param mixed $value
+     *
+     * @return string
+     */
+    protected function fixupFloat($value)
```

Comment:
Copied from Solr actually, now inlined in https://github.com/ezsystems/ezpublish-kernel/pull/1697/commits/16d6b2010a80ffad06d9e45963b61dfa410fb9c1.